### PR TITLE
Manual calibration

### PR DESCRIPTION
**Pull request by mirceacaprioru**
This is a pull request for merging the manual calibration tool. This tool is hidden and manual_calibration property must be enabled inside .ini file. 

Also in order to calibrate the power supply m2k_calib.ini must be created on m2k mass storage drive. 
Once the calibration procedures are executed m2k_calib.ini must be overwritten and m2k ejected in order to write the new calibration values into the context attributes in IIO. 

For the waveform generator and oscilloscope the automatic calibration is still in place and is executed at scopy startup.

### CODE REVIEW

**Comment by adisuciu** [comment]
Please fix conflicts and resubmit.

File: src/manualcalibration.h [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+	Calibration *calib;
+
+	struct stCalibStory stCalibrationStory;
+	struct stCalibParam stParameters;
+	QStringList calibListString;
+	QMap<QString, int> calibOption;
+
+	/*Calibrations procedure stories*/
+	const QStringList positiveOffsetStory = (QStringList() <<
```
Comment (dNechita) [inline src/manualcalibration.h]: Also, can be static const in the .cpp file. There's no need for a string list, a simple string will do.
Here's a nice way to declare a text block that contains line endings, spaces, tabs, etc.
```
QString myString(R"(
A text block with line endings
   and
    spaces
   and
    tabs
)");
```
More about raw string literal can be found here: http://en.cppreference.com/w/cpp/language/string_literal

File: src/tool_launcher.hpp [hunk 2]
```diff
@@ -244,6 +250,9 @@ class ToolLauncher_API: public ApiObject
 	bool debugger_enabled();
 	void enable_debugger(bool);
 
+	bool manual_calibration_enabled();
```
Comment (dNechita) [inline src/tool_launcher.hpp]: Method can be const.

File: src/manualcalibration.h (hunk 1 above)
Comment (mirceacaprioru) [reply src/manualcalibration.h]: These are QStringList because they hold multiple steps. The stories for these steps must be separated in some way.

Comment (dNechita) [reply src/manualcalibration.h]: Ok, it makes sense.

**Pull request merged**